gobblin-data-management/src/main/java/gobblin/data/management/copy/watermark/CopyEntityWatermarkGenerator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy.watermark;
+
+import java.io.IOException;
+
+import com.google.common.base.Optional;
+
+import gobblin.data.management.copy.CopyEntity;
+import gobblin.source.extractor.ComparableWatermark;
+import gobblin.source.extractor.WatermarkInterval;
+
+
+/**
+ * Watermark generator for {@link CopyEntity}.
+ */
+public interface CopyEntityWatermarkGenerator {
+  /**
+   * Generate optional {@link WatermarkInterval} for a given {@link CopyEntity}.
+   */
+  public Optional<WatermarkInterval> generateWatermarkIntervalForCopyEntity(CopyEntity copyEntity) throws IOException;
```

Comment:
Not for my case. I cannot really come up a good use case for other `CopyEntity`. Will remove that.